### PR TITLE
mjpg-streamer: update to last upstream version

### PR DESCRIPTION
**Pull request by ja-pa**
Maintainer: @thess
Compile tested: Turris Omnia (TOS4), OpenWrt 18.06.2
Run tested: Turris Omnia (TOS4), OpenWrt 18.06.2

Description:
This PR updates mjpg-streamer to upstream version and adds option listen_ip, which enables specify bind IP address.

Signed-off-by: Jan Pavlinec <jan.pavlinec@nic.cz>

### CODE REVIEW

**Comment by diizzyy** [comment]
I wonder if it would be better idea to import ustreamer instead and possibly libjpeg-turbo with it.
https://github.com/pi-kvm/ustreamer
https://libjpeg-turbo.org/

**Comment by ja-pa** [comment]
@diizzyy Do you mean to drop mjpg-streamer and create the package for uStreamer ? If so, I think that mjpg-streamer has more features and I don't think that it could fully replace mjpg-streamer. But I can imagine a separate uStreamer package for quality image streaming.

**Comment by diizzyy** [comment]
@ja-pa 
Yes, the only lacking feature (as far as I can tell) is "Access to webcam controls (focus, servos)and settings such as brightness via HTTP" which I'm unsure how relevant and functional it is today.

**Comment by ja-pa** [comment]
@diizzyy Well, there are still different input plugins and file output plugin (and luci app). But it's sufficient replacement if you use only http output plugin.

**Comment by thess** [comment]
I am not in favor of removing mjpg-streamer from this repo. It is far too popular and useful to do so. uStreamer discussion is not appropriate here.

btw - I have pan/tilt controls working with my ol Logitech Orb. You need to include libv4l2 IIRC,

File: multimedia/mjpg-streamer/files/mjpg-streamer.config [hunk 1]
```diff
@@ -11,5 +11,6 @@ config mjpg-streamer 'core'
 	option led 'auto'
 	option www '/www/webcam'
 	option port '8080'
+	option listen_ip '192.168.1.1'
```
Comment (thess) [inline multimedia/mjpg-streamer/files/mjpg-streamer.config]: Default should be **none** - not all routers use this subnet.

File: multimedia/mjpg-streamer/files/mjpg-streamer.init [hunk 2]
```diff
@@ -67,14 +67,18 @@ start_instance() {
 		output_arg="output_http.so"
 
 		config_get port "$s" 'port'
-		[ -n "$port" ] && output_arg="${output_arg} --port $port"
+		[ -n "$port" ] && output_arg="${output_arg} -p $port"
```
Comment (thess) [inline multimedia/mjpg-streamer/files/mjpg-streamer.init]: This is gratuitous and unnecessary

File: multimedia/mjpg-streamer/files/mjpg-streamer.init [hunk 3]
```diff
@@ -67,14 +67,18 @@ start_instance() {
...
+
+		config_get listen_ip "$s" 'listen_ip'
+		[ -n "$listen_ip" ] && output_arg="${output_arg} -l $listen_ip"
 
 		config_get www "$s" 'www'
-		[ -n "$www" ] && output_arg="${output_arg} --www $www"
+		[ -n "$www" ] && output_arg="${output_arg} -w $www"
```
Comment (thess) [inline multimedia/mjpg-streamer/files/mjpg-streamer.init]: Likewise unnecessary

File: multimedia/mjpg-streamer/files/mjpg-streamer.init [hunk 4]
```diff
@@ -67,14 +67,18 @@ start_instance() {
...
 
 		config_get username "$s" 'username'
 		config_get password "$s" 'password'
-		[ -n "$username" ] && [ -n "$password" ] && output_arg="${output_arg} --credentials $username:$password"
+
+		[ -n "$username" ] && [ -n "$password" ] && output_arg="${output_arg} -c $username:$password"
```
Comment (thess) [inline multimedia/mjpg-streamer/files/mjpg-streamer.init]: My preference is to use the long (and un-ambiguous) names

**Comment by thess** [comment]
@ja-pa - I don't see any need for so many non-functional changes to the patch files other than adding support for zmqserver which I don't know if we need.

How about support for building raspicam - that is useful!

File: multimedia/mjpg-streamer/files/mjpg-streamer.config (hunk 1 above)
Comment (ja-pa) [reply multimedia/mjpg-streamer/files/mjpg-streamer.config]: Fixed, the default is now commented out.

File: multimedia/mjpg-streamer/files/mjpg-streamer.init (hunk 2 above)
Comment (ja-pa) [reply multimedia/mjpg-streamer/files/mjpg-streamer.init]: Changed back.

File: multimedia/mjpg-streamer/files/mjpg-streamer.init (hunk 3 above)
Comment (ja-pa) [reply multimedia/mjpg-streamer/files/mjpg-streamer.init]: Changed back.

File: multimedia/mjpg-streamer/files/mjpg-streamer.init (hunk 4 above)
Comment (ja-pa) [reply multimedia/mjpg-streamer/files/mjpg-streamer.init]: Changed back

**Comment by ja-pa** [comment]
@thess Thanks for comments. I changed PR accordingly. And in case of patch file changes, I made it because otherwise, it ends up with rejection (020-remove-auto-lib-selection.patch )

**Comment by neheb** [comment]
Concerns seem to have been addressed. Merging.

**Pull request merged**